### PR TITLE
tests: kernel: Move tests/kernel/profiling to tests/kernel/threads

### LINKED ISSUE #7966: Move k_thread_foreach() tests to tests/kernel/threads
Please move the tests from `tests/kernel/profiling/` to where it belongs, under thread tests.

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Move tests/kernel/profiling test to tests/kernel/threads folder.

Fixes #7966

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

**Review by nashif — CHANGES_REQUESTED** [review]
thats not what the issue is about, did not asking for moving the test. I am asking for testing the k_thread_foreach functions on the API level as part of thread testing and not as part of getting stack information listed. Those are two different things

**Comment by ramakrishnapallala** [comment]
@nashif , I will move the test_k_thread_foreach test to threads_api tests.

File: tests/kernel/threads/thread_apis/prj.conf [hunk 1]
```diff
@@ -1,4 +1,6 @@
 CONFIG_ZTEST=y
 CONFIG_THREAD_MONITOR=y
-CONFIG_HEAP_MEM_POOL_SIZE=256
 CONFIG_THREAD_CUSTOM_DATA=y
+CONFIG_INIT_STACKS=y
```
Comment (andrewboie) [inline tests/kernel/threads/thread_apis/prj.conf]: why is CONFIG_INIT_STACKS enabled? does the test use it?

Comment (ramakrishnapallala) [reply tests/kernel/threads/thread_apis/prj.conf]: we don't need this in this test. I will remove.

**Comment by nashif** [comment]
recheck

**Comment by nashif** [comment]
recheck

**Comment by nashif** [comment]
recheck

**Review by nashif — APPROVED** [review]

**Pull request merged**